Makefile

```diff
@@ -577,40 +577,82 @@ distclean: clean
 html: $(HTML)
 
 # Rules to attach JavaScript and CSS assets to the head
-search/index.html: search/index.html.in .make-asset-block .make-template-block
+search/index.html: .make-asset-block .make-template-block
 	sed -e '/%ASSETS%/ {' -e 'r .make-asset-block' -e 'd' -e '}' \
 		-e '/%TEMPLATES%/ {' -e 'r .make-template-block' -e 'd' -e '}' \
-		search/index.html.in > search/index.html
+		search/search.html.in > search/index.html
 
-login/index.html: login/index.html.in .make-asset-block
+login/index.html: .make-asset-block
 	sed -e '/%ASSETS%/ {' -e 'r .make-asset-block' -e 'd' -e '}' \
-		login/index.html.in > login/index.html
-
-detailed/index.html: detailed/index.html.in .make-detailed-asset-block .make-detailed-template-block
+		-e 's/%APPNAME%//' \
+		-e 's/%PAGE%/Chaise :: Login/' \
+		-e 's/%NAVBAR%//' \
+		-e 's/%FOOTER%//' \
+		-e 's/%TEMPLATES%//' \
+		-e '/%CONTENT%/ {' -e 'r login/login.html.in' -e 'd' -e '}' \
+		index.html.in > login/index.html
+
+viewer/index.html: .make-viewer-asset-block
```

Comment:
these blocks all repeat... seems like they could be collapsed into a single pattern based rule?